R/convertEpochData.R

```diff
@@ -0,0 +1,292 @@
+          colnames(D) = as.character(colnames)[1:ncol(D)]
+          # ! Assumptions about columns names
+          colnames(D) = gsub(pattern = "datum|date", replacement = "date", x = colnames(D), ignore.case = TRUE)
+          colnames(D) = gsub(pattern = "tijd|time", replacement = "time", x = colnames(D), ignore.case = TRUE)
+          colnames(D) = gsub(pattern = "activiteit|activity", replacement = "ZCY", x = colnames(D), ignore.case = TRUE)
+          
+          timestamp_POSIX = as.POSIXct(x = paste(D$date[1:4], D$time[1:4], sep = " "), format = "%d-%m-%Y %H:%M:%S", tz = tz)
+          epSizeShort = mean(diff(as.numeric(timestamp_POSIX)))
+          if (epSizeShort != params_general[["windowsizes"]][1]) {
+            stop(paste0("\nThe short epoch size as specified by the user as the first value of argument windowsizes (",
+                 params_general[["windowsizes"]][1],
+                 " seconds) does NOT match the short epoch size we see in the data (", epSizeShort),
+                 " seconds). Please correct.", call. = FALSE)
+          }
+          timestamp_POSIX = timestamp_POSIX[1]
+          D = D[, "ZCY"]
+        } else if (params_general[["dataFormat"]] == "actiwatch_awd") {
+          # ! Assumption that first data row equals the first row with 3 columns
+          ind = 0
+          NC = 1
+          while (NC >= 3) {
+            testraw = data.table::fread(input = fnames[i],
+                                      header = FALSE, sep = ",", skip = ind,
+                                      nrows = 1, data.table = TRUE)
+            NC = ncol(testraw)
+            if (NC >= 3) {
+              break()
+            } else {
+              ind = ind + 1
+            }
+          }
+          D = data.table::fread(input = fnames[i],
+                                header = FALSE, sep = ",", skip = ind)
+          D = D[,1]
+          colnames(D)[1] = "ZCY"
+          header = data.table::fread(input = fnames[i],
+                                header = FALSE, sep = ",", nrows =  7)
+          # Get epoch size
+          optionalEpochs = data.frame(code = c("1", "2", "4", "8", "20", "81", "C1", "C2"),
+                                      size = c(15, 30, 60, 120, 300, 2, 5, 10))
+          epSizeShort = optionalEpochs$size[which(optionalEpochs$code == as.character(header[4]))]
+          # Get starttime 
+          timestamp_POSIX = as.POSIXct(x = paste(header[2], header[3], sep = " "),
+                                       format = "%d-%b-%Y %H:%M", tz = tz)
+          if (epSizeShort != params_general[["windowsizes"]][1]) {
+            stop(paste0("\nThe short epoch size as specified by the user as the first value of argument windowsizes (",
+                        params_general[["windowsizes"]][1],
+                        " seconds) does NOT match the short epoch size we see in the data (", epSizeShort),
+                 " seconds). Please correct.", call. = FALSE)
+          }
```

Comment:
Maybe in the future we might allow for reintegration to longer epoch sizes? For example, if file is provided in 15 seconds, it would be possible to reintegrate that to 60.

Reply:
Good idea, but needs more research to implement as we do not always know whether aggregation is to sum or to average.